nomad/structs/structs.go

```diff
@@ -1361,9 +1361,19 @@ type NodeUpdateResponse struct {
 	// region.
 	Servers []*NodeServerInfo
 
+	// ClientStatus is used to inform clients what the server-side
+	// has for their scheduling status during heartbeats.
+	ClientStatus *ClientStatus
```

Comment:
`SchedulingEligibility string` would be fine here. Would just need to check for `SchedulingEligibility != ""` instead of `ClientStatus != nil`.